### PR TITLE
determine correct edit form

### PR DESCRIPTION
**Pull request by abe-phdev**

### CODE REVIEW

**Review by Don-PhDev — APPROVED** [review]

**Review by Don-PhDev — APPROVED** [review]

File: app/views/incomes/_form.html.erb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    <div id="error_explanation">
+      <h2><%= pluralize(@income.errors.count, "error") %> prohibited this transaction from being saved:</h2>
+
+      <ul>
+        <% @income.errors.each do |error| %>
+          <li><%= error.full_message %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
```
Comment (rusbal) [inline app/views/incomes/_form.html.erb]: Create a shared partial and move this part of the view.
Path: `app/view/shared/_form_errors.html.erb`

Call like this:
```
<%= render 'shared/form_errors', object: @income %>
```

Inside the partial:
```
<% if object.errors.any? %>
    <div id="error_explanation">
    ...
```

Comment (rusbal) [reply app/views/incomes/_form.html.erb]: Then refactor other forms as well.

File: app/views/incomes/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  </div>
+
+  <div class="field">
+    <%= form.label :due_date %>
+    <%= form.date_field :due_date %>
+  </div>
+
+  <div class="field">
+    <%= form.label :actualized_at %>
+    <%= form.date_field :actualized_at, min: Date.today - 1000, max: DateTime.now %>
```
Comment (rusbal) [inline app/views/incomes/_form.html.erb]: Change to 
```
min: 30.days.ago
```

File: app/views/salary_expenses/_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  </div>
+
+  <div class="field">
+    <%= form.label :due_date %>
+    <%= form.date_field :due_date %>
+  </div>
+
+  <div class="field">
+    <%= form.label :actualized_at %>
+    <%= form.date_field :actualized_at, min: Date.today - 1000, max: DateTime.now %>
```
Comment (rusbal) [inline app/views/salary_expenses/_form.html.erb]: Change to

`min: 30.days.ago`

File: app/views/transactions/_tr.html.erb [hunk 4]
```diff
@@ -7,6 +7,12 @@
   <td><%= tran.due_date %></td>
   <td><%= tran.actualized? ? tran.actualized_at : "Draft" %></td>
   <td><%= link_to 'Show', tran %></td>
-  <td><%= link_to 'Edit', edit_transaction_path(tran) unless tran.actualized? %></td>
+  
+  <% if tran.transaction_type.name == TransactionType.income.name %>
```
Comment (rusbal) [inline app/views/transactions/_tr.html.erb]: Change to
```
if tran.transaction_type.income?
```

In `TransactionType` model
```
def income?
  name == INCOME_PROGRAMMING
end
```

File: app/views/transactions/_tr.html.erb [hunk 5]
```diff
@@ -7,6 +7,12 @@
...
+    <td><%= link_to 'Edit', edit_income_path(tran) unless tran.actualized? %></td>
+  <% elsif tran.transaction_type.name == TransactionType.salary.name %>
+    <td><%= link_to 'Edit', edit_salary_expense_path(tran) unless tran.actualized? %></td>
```
Comment (rusbal) [inline app/views/transactions/_tr.html.erb:14]: Do the same for the following code as well
```
tran.transaction_type.name == TransactionType.salary.name
```

**Review by rusbal — APPROVED** [review]
🚢 :shipit:

**Pull request merged**